src/state.py

```diff
@@ -5,11 +5,54 @@ def __init__(self):
     def add_student(self, student):
         self.students.append(student)
 
+    #Given two students and a subject, trades their classes for that given subject
+    def trade_classes(self,student1,student2,subject_name):
+        student1_class = student1.get_class_for_subject(subject_name)
+        student2_class = student2.get_class_for_subject(subject_name)
+        student1.set_class_for_subject(subject_name, student2_class)
+        student2.set_class_for_subject(subject_name, student1_class)
+        #student1.remove_subject_target(subject_name) #E suposto ele remover os targets a medida que vai trocando? Ou e usado para a funçao de fitness?
+        #student2.remove_subject_target(subject_name)
+
+    def generate_neighbour(self):
+        for student_index,student in enumerate(self.students,start=1): # select a student
+
+
+            for subject_name,target_classes in student.subject_targets.items(): # select a subject name and its respective  target classes
+
+                student_class = student.get_class_for_subject(subject_name) # get the current class of the first student for that subject
+
+
+                for target_class in target_classes: # select, for that subject, a possible target class
+
+                    for trader_student in self.students[student_index:]: # find a trader student
+                        if trader_student == student:
+                            continue
+                        
+                        trader_class = trader_student.get_class_for_subject(subject_name) # get the class of the trader student, for that subject
+
+                        if trader_class is None or trader_class != target_class: # if target student does not have the desired class for that subject, continue
+                            continue
+                  
+                        trader_targets = trader_student.get_targets_for_subject(subject_name) # get the targets for that subject, for the trader student
+                        
+                        if trader_targets is not None and student_class in trader_targets:
+                            self.trade_classes(student,trader_student,subject_name) #trades the students classes for that subject
+                            break
```

Comment:
Do you really want to break the loop only, or stop the function with a `return` instead?